### PR TITLE
How we chose react

### PR DESCRIPTION
**Pull request by richardbatty**
I'd like some feedback on this, it seems pretty boringly written.

### CODE REVIEW

**Comment by reevoo-samuel** [comment]
*If you're new, read the [guidelines](https://github.com/reevoo/guidelines/blob/master/pull_requests.md).*

Good code is **tested**, **easy to understand** and **able to cope when things go wrong**.

Good pull requests are **descriptive**, **small** and **short-lived**.

**For the Author**
- [ ] Description explains what you have done and why.
- [ ] Tests have been written (or you have explained why you don't need tests).
- [ ] Documentation/comments have been added/updated.

**For the Reviewer**
- [ ] Tests are green on CI (or pulled/tested on your machine).
- [ ] Tests cover happy and unhappy paths in the code.
- [ ] The code is easy to understand.

File: _posts/2015-09-22-how-we-chose-react.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+# How we chose React
+
+In a recent project to build a comments widget, we decided to experiment with different approaches and frameworks. This post explains what we did and how we ended up choosing [React](https://facebook.github.io/react/).
+
+## What we didn't want
+
+### Angular
+
+We knew we didn't like Angular from previous experience. Firstly, it took ages to learn - there are lots of unnecessarily confusing parts like the differences between service, factory, and provider or what @, &, and = mean in directives. We also didn't like its templates. Having a special templating language is more to learn, and even worse, it is hard to debug. When you cause an error in your template, you get a stack trace that doesn't include any of your code, making it hard to find the problem. And because it's not JavaScript, you can't put a debugger in to see what's going on. Finally, we found the testing very time consuming. There was a lot of error-prone boilerplate that we had to write in the ```beforeEach``` of every set of tests. Overall it felt like we had to warp our brains into Angular's world, rather than the framework helping us and then getting out of the way.
```
Comment (jonnyarnold) [inline _posts/2015-09-22-how-we-chose-react.md]: Top tip - for inline code, you only need single backticks `like so`

File: _posts/2015-09-22-how-we-chose-react.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+As we were rewriting the app in React, we found it a pleasure to use. For example, we had a photo uploading component within the widget which had to be displayed in several different states depending on whether there were no photos or some photos, whether the photo upload limit had been reached or not, and whether a photo was currently being uploaded or not. We decided to create objects that defined each photo uploader state and then rerender the component every time the state changed. Rewriting the component this way was extremely easy despite the multiple state changes going on. This shows the power of programming state change explicitly and separating logic code from templating code.
+
+## What we learned
+
+The most important thing we learned was about how to deal with problems: If you're having problems with your current approach, stop and solve the problem. Don't just plough on thinking that going back will take too much time. It will take you more time fighting with a solution that doesn't work properly and will lead to a worse result.
```
Comment (jonnyarnold) [inline _posts/2015-09-22-how-we-chose-react.md]: I need to take this paragraph, put it in bold and tape it to my monitor.

**Comment by jonnyarnold** [comment]
You need to run `bundle exec rake jekyll:serve` to get a Soapbox Content ID. (See *Getting your post live* at the bottom of the readme.) Otherwise, LGTM

**Pull request merged**